modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/SearchPermissionCheckerImpl.java

```diff
@@ -622,10 +581,13 @@ private String _getPermissionName(
 	private static final Log _log = LogFactoryUtil.getLog(
 		SearchPermissionCheckerImpl.class);
 
-	private final Collection<SearchPermissionFieldContributor>
-		_searchPermissionFieldContributors = new CopyOnWriteArrayList<>();
-	private final Collection<SearchPermissionFilterContributor>
-		_searchPermissionFilterContributors = new CopyOnWriteArrayList<>();
+	@Reference(policyOption = ReferencePolicyOption.GREEDY)
+	private volatile List<SearchPermissionFieldContributor>
```

Comment:
@W2jx It seems both `SearchPermissionFieldContributor` and `SearchPermissionFilterContributor` are referenced here and in the other 2 classes you modified. Please extract them to a new component to avoid duplicate service trackers. Please also double check other PRs you sent. Thanks!